bundles/org.openhab.binding.modbus.stiebeleltron/src/main/java/org/openhab/binding/modbus/stiebeleltron/internal/handler/StiebelEltronHandler.java

```diff
@@ -0,0 +1,713 @@
+     * @param command get the value of this command.
+     * @return short the value of the command as short
+     */
+    private short getInt16Value(Command command) throws StiebelEltronException {
+        if (command instanceof DecimalType) {
+            DecimalType c = (DecimalType) command;
+            return c.shortValue();
+        }
+        throw new StiebelEltronException("Unsupported command type");
+    }
+
+    /**
+     * Handle incoming commands.
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (RefreshType.REFRESH == command) {
+            String groupId = channelUID.getGroupId();
+            if (groupId != null) {
+                AbstractBasePoller poller;
+                switch (groupId) {
+                    case GROUP_SYSTEM_STATE:
+                        poller = systemStatePoller;
+                        break;
+                    case GROUP_SYSTEM_PARAMETER:
+                        poller = systemParameterPoller;
+                        break;
+                    case GROUP_SYSTEM_INFO:
+                        poller = systemInformationPoller;
+                        break;
+                    case GROUP_ENERGY_INFO:
+                        poller = energyPoller;
+                        break;
+                    default:
+                        poller = null;
+                        break;
+                }
+                if (poller != null) {
+                    poller.poll();
+                }
+            }
+        } else {
+            try {
+                if (GROUP_SYSTEM_PARAMETER.equals(channelUID.getGroupId())) {
+                    switch (channelUID.getIdWithoutGroup()) {
+                        case CHANNEL_OPERATION_MODE:
+                            writeInt16(1500, getInt16Value(command));
+                            break;
+                        case CHANNEL_COMFORT_TEMPERATURE_HEATING:
+                            writeInt16(1501, getScaledInt16Value(command));
+                            break;
+                        case CHANNEL_ECO_TEMPERATURE_HEATING:
+                            writeInt16(1502, getScaledInt16Value(command));
+                            break;
+                        case CHANNEL_COMFORT_TEMPERATURE_WATER:
+                            writeInt16(1509, getScaledInt16Value(command));
+                            break;
+                        case CHANNEL_ECO_TEMPERATURE_WATER:
+                            writeInt16(1510, getScaledInt16Value(command));
+                            break;
+                    }
+                }
+            } catch (StiebelEltronException error) {
+                if (hasConfigurationError() || getThing().getStatus() == ThingStatus.OFFLINE) {
+                    return;
+                }
+                String msg = "";
+                String cls = "";
+                cls = error.getClass().getName();
+                msg = error.getMessage();
```

Comment:
Initialization with empty strings has no effect, since they are overwritten below. You can initialize them with the intended values at once.

Reply:
Removed the initialization with the empty strings